pkg/server/handler.go

```diff
@@ -88,35 +110,37 @@ func (mc *MsgHandler) subscribeToMessages(msgChan chan network.InMsg) error {
 	return err
 }
 
-func (mc *MsgHandler) onMsgReceived(msgChan chan network.InMsg) {
-	for {
-		var err error
-		msg := <-msgChan
-		mc.logger.Infof("exchange: %s, routing key: %s", msg.Exchange, msg.RoutingKey)
-		mc.logger.Infof("message received: %s", string(msg.Body))
-
-		token, ok := msg.Headers["Authorization"].(string)
-		if !ok {
-			mc.logger.Error(errors.New("authorization token not provided"))
-			continue
-		}
+func (mc *MsgHandler) onMsgReceived(msgChan chan network.InMsg) error {
+	msg := <-msgChan
+	var err error
```

Comment:
Have you tried this? :smile: 

```suggestion
func (mc *MsgHandler) onMsgReceived(msgChan chan network.InMsg) err error {
	msg := <-msgChan
```

Reply:
Nice found!